### ISSUE TITLE
azurerm_app_service_certificate must be recreated for no reason

### ISSUE THREAD

**Issue by GurliGebis** [hashicorp/terraform-provider-azurerm#18652]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.1

### AzureRM Provider Version

3.25.0

### Affected Resource(s)/Data Source(s)

azurerm_app_service_certificate

### Terraform Configuration Files

```hcl
https://www.dropbox.com/s/3apku9aig88yikw/terraform-issue.zip?dl=0
```

### Debug Output/Panic Output

```shell
https://www.dropbox.com/s/bzenz4isbut0sob/log.txt.gpg?dl=0

Log file, encrypted with GPG
```

### Expected Behaviour

I changed the key_permissions on the managed_identity resource inside the keyvault.tf file from:
key_permissions = [ "Get", "List" ]
to:
key_permissions = [ "Get", "List", "Delete" ]

This should just have caused it to change the access policy.

### Actual Behaviour

I wants to recreate several resources - I have included a cut down output from "terraform plan" here:

Terraform will perform the following actions:

* module.apps.data.azurerm_key_vault.keyvault will be read during apply
* module.apps.data.azurerm_resource_group.resourcegroup will be read during apply
* module.apps.data.azurerm_user_assigned_identity.managed_identity will be read during apply
* module.apps.azurerm_app_service_certificate.certificate **must be replaced**
* module.apps.azurerm_app_service_custom_hostname_binding.redirect **must be replaced**
* module.apps.azurerm_key_vault_certificate.certificate_website **must be replaced**
* module.apps.azurerm_key_vault_secret.dummy **must be replaced**
* module.apps.azurerm_linux_web_app.app_service **must be replaced**
* module.apps.azurerm_service_plan.webapp **must be replaced**
* module.keyvault.azurerm_key_vault_access_policy.managed_identity will be **updated in-place**

### Steps to Reproduce

1. Edit main.tf and deside on a suffix
2. Setup DNS for your custom domain, so Azure can set up a custom domain for it.
3. Create a selfsigned certificate for it, save it as a pfx file called "cert.pfx" with the password "Test1234"
4. Create a resource group called "rg-YOUR-SUFFIX"
5. Run terraform apply to set up the site
6. Once setup is done, change the "key_permissions" property for the "azurerm_key_vault_access_policy" inside "keyvault.tf" called "managed_identity" - just add the "Delete" permission to the array.
7. Run terraform plan

Now it wants to recreate way too many resources, even though it should just update the access policy.

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v3.x of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
